File: src/burp/BurpExtender.java
package burp;

import javax.swing.*;
import java.awt.*;

public class BurpExtender implements IBurpExtender, IMessageEditorTabFactory {

    private IBurpExtenderCallbacks callbacks;
    private IExtensionHelpers helpers;

    @Override
    public void registerExtenderCallbacks(IBurpExtenderCallbacks callbacks) {
        this.callbacks = callbacks;
        this.helpers = callbacks.getHelpers();
        callbacks.setExtensionName("InstaTools");
        callbacks.registerMessageEditorTabFactory(this);
    }

    @Override
    public IMessageEditorTab createNewInstance(IMessageEditorController controller, boolean editable) {
        return new IgMessageTab(controller, editable, callbacks, helpers);
    }
}

File: src/burp/BurpExtender.kt
package burp

import de.nerixyz.insta.IgMessageTab
import de.nerixyz.insta.decoders.IgApiDecoder
import de.nerixyz.insta.decoders.IgGraphDecoder

class BurpExtender : IBurpExtender, IMessageEditorTabFactory {
    private var callbacks: IBurpExtenderCallbacks? = null
    private var helpers: IExtensionHelpers? = null
    override fun registerExtenderCallbacks(callbacks: IBurpExtenderCallbacks) {
        this.callbacks = callbacks
        helpers = callbacks.helpers
        callbacks.setExtensionName("InstaTools")
        callbacks.registerMessageEditorTabFactory(this)
    }

    override fun createNewInstance(controller: IMessageEditorController, editable: Boolean): IMessageEditorTab {
        val callbacks = this.callbacks
        if (callbacks == null || callbacks.helpers == null) {
            throw IllegalStateException("Callbacks or helpers are null")
        }
        return IgMessageTab(controller, editable, callbacks, callbacks.helpers, arrayListOf(IgApiDecoder(), IgGraphDecoder()))
    }
}